com.unity.render-pipelines.high-definition/Editor/BuildProcessors/HDRPPreprocessShaders.cs

```diff
@@ -483,79 +500,97 @@ public void Dispose()
-                var hdPipelineAssets = ShaderBuildPreprocessor.hdrpAssets;
+                // Call list of strippers
+                // Note that all strippers cumulate each other, so be aware of any conflict here
+                if (shaderProcessorsList
+                    .Any(shaderPreprocessor => shaderPreprocessor.ShadersStripper(hdAsset, shader, snippet, input)))
+                    continue;
+
+                removeInput = false;
+                break;
+            }
 
-                if (hdPipelineAssets.Count == 0)
-                    return;
+            return removeInput;
+        }
 
-                uint preStrippingCount = (uint)inputData.Count;
+        public void OnProcessShader(Shader shader, ShaderSnippetData snippet, IList<ShaderCompilerData> inputData)
+        {
+            if (inputData == null || inputData.Count == 0)
+                return;
 
-                // Test if striping is enabled in any of the found HDRP assets.
-                if (hdPipelineAssets.Count == 0 || !hdPipelineAssets.Any(a => a.allowShaderVariantStripping))
-                    return;
+            uint preStrippingCount = (uint)inputData.Count;
+            int outputVariantCount = inputData.Count;
 
-                var inputShaderVariantCount = inputData.Count;
-                for (int i = 0; i < inputShaderVariantCount;)
-                {
-                    ShaderCompilerData input = inputData[i];
+            if (!active)
+            {
+                if (!IsShaderProcessed(shader, snippet))
+                    return;
 
-                    // Remove the input by default, until we find a HDRP Asset in the list that needs it.
-                    bool removeInput = true;
+                outputVariantCount = 0;
+            }
 
-                    foreach (var hdAsset in hdPipelineAssets)
-                    {
-                        var strippedByPreprocessor = false;
-
-                        // Call list of strippers
-                        // Note that all strippers cumulate each other, so be aware of any conflict here
-                        foreach (BaseShaderPreprocessor shaderPreprocessor in shaderProcessorsList)
-                        {
-                            if (shaderPreprocessor.ShadersStripper(hdAsset, shader, snippet, input))
-                            {
-                                strippedByPreprocessor = true;
-                                break;
-                            }
-                        }
-
-                        if (!strippedByPreprocessor)
-                        {
-                            removeInput = false;
-                            break;
-                        }
-                    }
+            double stripTimeMs;
+            using (new ExportShaderStrip("Temp/shader-strip.json", shader, snippet, inputData, this))
+            {
+                Stopwatch shaderStripingWatch = new Stopwatch();
+                shaderStripingWatch.Start();
```

Comment:
[Optional] If this start to be used elsewhere too, think about making a MesuringScope